### PR TITLE
Sets the parent in insertBefore/After and refactors these methods.

### LINKED ISSUE #345: insertBefore does not set the parent

### PR DESCRIPTION
**Pull request by GerardPaligot**
Closes #345

### CODE REVIEW

**Comment by monperrus** [comment]
cool!

since you change the test file, it's impossible to see the new test case and assertion that you are fixing.

**Comment by GerardPaligot** [comment]
> since you change the test file, it's impossible to see the e new test case and assertion that you are fixing.

The formatting of this file was really (!) bad. Some test cases used tabs, other used spaces and even both sometimes. And I renamed the file.

But I didn't change any test, just add new assertions on existing tests and create new tests.

**Comment by monperrus** [comment]
I can imagine.

It's better to split the PR in two, one for the bug fix (with a clear
additional assertion) and another one for the formatting.

Might be two commits in the same PR.

File: src/main/java/spoon/support/reflect/code/CtStatementImpl.java [hunk 1]
```diff
@@ -55,18 +58,8 @@ public static void insertAfter(CtStatement target, CtStatementList statements)
...
-			if (s == target) {
-				break;
-			}
-		}
-		for (int j = statements.getStatements().size() - 1; j >= 0; j--) {
-			CtStatement s = statements.getStatements().get(j);
-			parentStatementList.getStatements().add(i, s);
-		}
+
+		new InsertBeforeVisitor(target, statements, InsertType.AFTER).scan(e);
```
Comment (GerardPaligot) [inline src/main/java/spoon/support/reflect/code/CtStatementImpl.java]: Change name to InsertVisitor

**Pull request merged**